elasticache-redis/main.tf

```diff
@@ -71,13 +71,65 @@ resource "aws_security_group" "redis" {
   tags = local.tags
 }
 
+resource "aws_kms_key" "ssm_redis_endpoint" {
+  description             = "KMS key for SSM parameters"
+  deletion_window_in_days = 10
+  enable_key_rotation     = true
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Id": "kms-key-ssm-redis-endpoint-ssl",
+  "Statement": [
+    {
+      "Sid": "Enable IAM User Permissions",
+      "Effect": "Allow",
+      "Principal": {
+        "AWS": "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      },
+      "Action": "kms:*",
+      "Resource": "*"
+    },
+    {
+      "Sid": "Allow SSM Use of the Key",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ssm.amazonaws.com"
+      },
+      "Action": [
+        "kms:Encrypt",
+        "kms:Decrypt",
+        "kms:ReEncrypt*",
+        "kms:GenerateDataKey*",
+        "kms:DescribeKey"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_ssm_parameter" "endpoint_short" {
```

Comment:
I would do this the other way round if this is the one we are going to keep long term. So call the other one `endpoint_deprecated` or something, leave this one as `endpoint`.

Reply:
@WillGibson Spoke with @acodeninja on this and he suggested keep it otherwise the terraform apply will delete the original one to recreate it with the new name.

Reply:
The end result will be the same though, and we won't get left having a resource with a funny name.

Reply:
Tested this with @tony-griffin and it causes some woes on the `terraform apply`. I retract my comments.